src/py21cmfast/wrapper/cfuncs.py

```diff
@@ -16,11 +17,82 @@
 
 logger = logging.getLogger(__name__)
 
+# Ideally, backend functions that we access here should do all the broadcasting/initialisation themselves
+# These decorators are for lower functions which are called directly in one or two lines, like delta_crit
+
+# TODO: a lot of these assume input as numpy arrays via use of .shape, explicitly require this
+
+
+def broadcast_params(func: Callable) -> Callable:
+    """Decorator to broadcast the parameters to the C library before calling the function."""
+
+    def wrapper(*args, **kwargs):
+        inputs = kwargs.get("inputs")
+        if inputs.astro_params:
```

Comment:
In our current implementation, `astro_params` is always set on `inputs`. It's never `None`. So this if/else is unnecessary